### ISSUE TITLE
Multi marker feature in a single map

### ISSUE THREAD

**Issue by pasangtamang** [ArnoldV/Our.Umbraco.GMaps#27]
Recently we were in need of adding multiple location. As a solution we were in need of option to add multi marker from in Google map plugin. We extended the feature from original package and now we would like to make this available for community. Below you can see how the UI is
![image](https://user-images.githubusercontent.com/45009265/96698164-8749de80-13ac-11eb-9640-cad2bd1d891a.png)
Next step is create a PR to merge with the plugin?

**Comment by ronaldbarendse** [comment]
@pasangtamang Rewriting this package to properly support multiple markers will require quite some work, so feel free to share the things you've already done by creating a PR.

The current editor is also explicitly named `Google Maps Single Marker` and the PVC only returns a single marker/address:
https://github.com/ArnoldV/Our.Umbraco.GMaps/blob/41786ebeea5b1201a4980e9d5d4fb1d833232eb7/src/Our.Umbraco.GMaps/App_Plugins/Our.Umbraco.GMaps/package.manifest#L5
https://github.com/ArnoldV/Our.Umbraco.GMaps/blob/41786ebeea5b1201a4980e9d5d4fb1d833232eb7/src/Our.Umbraco.GMaps.Core/PropertyValueConverter/GMapsPropertyValueConverter.cs#L13

To prevent breaking changes, the multiple marker editor can be created as a completely different editor or would otherwise require at least the following:
- Specify minimum and maximum amount of markers (similar to the Multi URL Picker), defaulting to maximum of 1 (to keep backwards compatibility when upgrading)
- Return a different value type in the PVC (either a list of `GmapsModel` or a new model class containing multiple addresses and a single map config, as that's the same for all addresses/markers)
- Store the property data differently (basically the same as with the PVC value, although this need to support existing data stored by Umbraco)

Because of all the complexities involved with this, I would recommend just adding this editor within a Nested Content/Block List and have every marker stored separately and render it in a single map on the front-end.

**Comment by pasangtamang** [comment]
@ronaldbarendse My implementation I have done as a separate data type. So existing single picker will still remain same as it was. I will add a PR so you can see the changes.

**Comment by robertjf** [comment]
@pasangtamang are you still interested in contributing your enhancements to the package? We've recently upgraded to support both Umbraco 8 & 9 and I'd be very interested to see your changes and how we can incorporate them...

**Comment by pasangtamang** [comment]
@robertjf It's been a long time I made changes as per few of my project requirement. I will check my repo and refresh myself. Then I will get back to you on this topic before end of this week.

**Comment by pasangtamang** [comment]
@robertjf I did a look into my code and found that due to new structural change in this package, I have to do some bug fixing. This will take me few more days. With the older version it works fine. If you want to check before my bug fixing in newer version then you can check in branch v1.3.2 from my repository (https://github.com/pasangtamang/Our.Umbraco.GMaps/tree/v1.3.2). There is one thing you will notice quickly. In backoffice you will find all the marker not showing at once. It switched only after you click on edit button.

**Comment by robertjf** [comment]
thanks @pasangtamang - I'll take a look and see how I can adapt it...

Ta
